lib/models/application.coffee

```diff
@@ -95,19 +95,45 @@ getApplicationModel = (deps, opts) ->
 	exports.getAll = (options = {}, callback) ->
 		callback = findCallback(arguments)
 
-		auth.getUserId()
-		.then (userId) ->
+		Promise.props(
+			userId: auth.getUserId()
+			organizationId: auth.getUserOrganizationId()
+		)
+		.then ({ userId, organizationId }) ->
 			return pine.get
 				resource: 'application'
 				options:
 					mergePineOptions
 						$orderby: 'app_name asc'
 						$filter:
-							$or:
-								user: userId
-								includes__user: $any:
-									$alias: 'x'
-									$expr: x: user: userId
+							# TODO: revisit when the API is finalized
+							# so that it reflects the code used on the
+							# my_applications custom resource
+							$or: [
+								organization: organizationId
+							,
+								organization: $any:
+									$alias: 'o'
+									$expr: o: user__is_member_of__organization: $any:
+										$alias: 'umo'
+										$expr: $and: [
+											umo: user: userId
+										,
+											umo: grants__organization_membership_role: name: $in: [
+												'owner'
+												'administrator'
```

Comment:
I'm surprised by this - why doesn't `application.getAll()` return all applications I own or can access, as it does in MC?

Reply:
Ah wait, I think I'm wrong. Let me confirm: it does do that, but we need an extra check for these roles, because we normally check access using the team check below, but owners & admins aren't in a team for the application. Right?

Reply:
Let me summarize my thinking and you can tell me if any of the reasoning steps is incorrect:
* As discussed back in that Arch call, the `sdk.applications.getAll()` is replicating the `my_applications` endpoint, which afaik is there so that resin admins do not get all the apps of the db in their dashboard.
* In the orgs era you can access an app because:
  * you own that app - aka it belongs to your personal org
  * it belongs to a team that you have access - being just a team member doesn't give you access to any app
  * you are an owner or admin of an org regardless of whether you are in any team of that org. The UX point is that I can add admins/owners and not have to add them to any/all teams to give them access to all apps. Checked the permissions while writing this See: https://github.com/resin-io/resin-api/pull/1079/files#diff-e4921799df81f37704fbab97e47687e3R30

We could possibly remove the `organization: organizationId`, but that was precautionary to avoid unpleasant situations.
Do all these make sense?